drivers/flash/soc_flash_nrf.c

```diff
@@ -475,7 +515,9 @@ static int erase_op(void *context)
 		}
 #endif /* CONFIG_SOC_FLASH_NRF_RADIO_SYNC */
 
-	} while (e_ctx->len > 0);
+	} while ((IS_ENABLED(CONFIG_SOC_FLASH_NRF_RADIO_SYNC) ||
+		  IS_ENABLED(CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE)) &&
+		  e_ctx->len > 0);
```

Comment:
I am still not convinced that this loop should not be continued when both radio synchronization and partial erase features are not enabled. User can still request to erase more than one page in a single call to `flash_erase`, and in such case `e_ctx->len` will be > 0 at this point but the loop will not iterate again. Or am I missing something?

Reply:
OK

Reply:
For some reason I thought it would be ok to let user limit flash erase operation to certain time only.